### PR TITLE
Add details about nuget.exe localizaiton

### PR DESCRIPTION
**Pull request by nkolev92**
Fixes https://github.com/NuGet/docs.microsoft.com-nuget/issues/2949

I wasn't able to find a way to reference to the localization of Visual Studio. 
If someone else can find it, I'd be happy to add it.

### CODE REVIEW

File: docs/reference/nuget-exe-CLI-Reference.md [hunk 1]
```diff
@@ -73,3 +73,8 @@ Developers concerned only with consuming packages, for example, need only unders
 
 > [!Note]
 > Command option names are case-insensitive. Options that are deprecated are not included in this reference, such as `NoPrompt` (replaced by `NonInteractive`) and `Verbose` (replaced by `Verbosity`).
+
+## Localization
+
+NuGet.exe's progress, warning and error messages are  translated in the same locales as Visual Studio.
```
Comment (erdembayar) [inline docs/reference/nuget-exe-CLI-Reference.md:79]: Should we mention about environmental var which controls it?
Maybe someone using non-english OS still wants to use English error message then we need to inform them how to do.

File: docs/reference/nuget-exe-CLI-Reference.md [hunk 2]
```diff
@@ -73,3 +73,8 @@ Developers concerned only with consuming packages, for example, need only unders
...
+NuGet.exe ships as a single exe, and due to size considerations, only the most commonly surface messages are translated in all languages.
```
Comment (donnie-msft) [inline docs/reference/nuget-exe-CLI-Reference.md:80]:
```suggestion
NuGet.exe ships as a single exe, and due to size considerations, only the most commonly surfaced messages are translated in all languages.
```

File: docs/reference/nuget-exe-CLI-Reference.md (hunk 1 above)
Comment (donnie-msft) [inline docs/reference/nuget-exe-CLI-Reference.md:79]:
```suggestion
NuGet.exe's progress, warning and error messages are translated into the same locales as Visual Studio.
```

Comment (nkolev92) [reply docs/reference/nuget-exe-CLI-Reference.md]: The environment variable is documented in https://learn.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-environment-variables. 

I don't expect this to be a super common scenario, so I don't think we need to add it.

**Review by jeffkl — APPROVED** [review]

**Pull request merged**